packages/react/src/components/header/components/headerActionBarItem/HeaderActionBarItem.module.scss

```diff
@@ -1,87 +1,156 @@
-@use '../../../../styles/common.scss';
-
-button.actionBarItem {
-  align-self: center;
-  appearance: none;
-  background: transparent;
-  border: none;
-  color: var(--header-color);
-  cursor: pointer;
-  padding: 0;
-
-  &:focus-visible {
-    border: none;
-    outline: var(--header-focus-outline-width) solid var(--header-focus-outline-color);
-    outline-offset: var(--header-focus-outline-width);
-  }
+.dropdownWrapper {
+  bottom: 0;
+  overflow: hidden;
+  position: absolute;
+  right: 0;
+  transform: translateY(100%) translateY(1px);
+  transition-duration: var(--animation-duration-dropwdown);
+  transition-property: max-height, padding-bottom;
+  transition-timing-function: ease;
+  z-index: 20;
 }
 
-.actionBarItemIcon {
-  display: block;
-  height: var(--icon-size);
-  margin: 0 auto;
-  padding: 3px 0 0;
-  pointer-events: none;
-  width: var(--icon-size);
+.container {
+  display: flex;
 
-  & > svg {
-    display: block;
-    height: 100%;
-    width: auto;
+  .dropdownWrapper {
+    max-height: 0;
+    min-width: 320px;
+    padding-bottom: 0;
+    transition-duration: var(--animation-duration-dropwdown);
+    transition-timing-function: ease;
   }
-}
 
-.actionBarItemLabel {
-  color: var(--action-bar-item-title-font-color);
-  display: block;
-  font-size: var(--action-bar-item-title-font-size);
-  margin: 0 auto;
-  pointer-events: none;
-  white-space: nowrap;
+  &.visible .dropdownWrapper {
+    max-height: 1000vh;
+    padding-bottom: 20px;
+  }
 
-  & > svg {
-    display: block;
+  &:not(.hasContent) {
+    display: none;
+  }
+
+  &:not(.fullWidth) {
+    display: flex;
+    position: relative;
+  }
+
+  &.fullWidth {
+    border-bottom: 1px solid var(--color-black-20);
+    display: flex;
+    justify-content: center;
+
+    > .dropdownWrapper {
+      left: 0;
+      position: relative;
+      transform: none;
+    }
+
+    &.hasSubItems {
+      display: block;
+    }
   }
-}
 
-.labelOnRight {
-  display: inline-block;
+  &.menuItem {
+    border-bottom: 0;
+  }
 }
 
-.fixedRightPosition {
+.icon {
   display: block;
 }
 
-@mixin hiddenButTakingHorizontalSpace {
-  height: 0;
-  opacity: 0;
-}
+.dropdown {
+  background: var(--header-background-color);
+  border: 1px solid var(--color-black-10);
+  border-top: 1px solid var(--color-black-20) !important;
+  margin-top: -1px;
+  position: relative;
+  right: 0;
+  transform: translateY(-100%);
+  transition:
+    transform var(--animation-duration-dropwdown) ease,
+    visibility var(--animation-duration-dropwdown) ease;
+  visibility: hidden;
+
+  &.fullWidth {
+    background: transparent;
+    border: 0;
+    left: 0;
+    margin: -2px;
+  }
 
-@mixin visibleAndTakingSpace {
-  height: auto;
-  opacity: 1;
-}
+  & > h3 {
+    font-size: 18px;
+    font-weight: bold;
+    line-height: 28px;
+    margin: 0;
+    padding: 16px;
+  }
 
-.activeStateContent {
-  @include hiddenButTakingHorizontalSpace;
+  & h4 {
+    padding: 12px 16px !important;
```

Comment:
Is there a `--spacing ` variable for these?

Reply:
And also `!important` should be used only as a last resort, is it?